File: Week_01-数组-链表-堆栈-队列/11.盛最多水的容器.cpp
/*
 * @lc app=leetcode.cn id=11 lang=cpp
 *
 * [11] 盛最多水的容器
 */

// @lc code=start
class Solution {
public:
    int maxArea(vector<int>& height) {
        int leftIndex = 0;
        int rightIndex = height.size() - 1;
        int maxVolum = 0;
        while (leftIndex < rightIndex) {
            maxVolum = max(maxVolum, (rightIndex - leftIndex) * min(height[leftIndex], height[rightIndex]));
            if (height[leftIndex] < height[rightIndex]) {
                leftIndex++;
            } else {
                rightIndex--;
            }
        }
        return maxVolum;
    }
};
// @lc code=end

File: Week_01-数组-链表-堆栈-队列/11.盛最多水的容器.py
#
# @lc app=leetcode.cn id=11 lang=python
#
# [11] 盛最多水的容器
#

# @lc code=start
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        n = len(height)
        leftindex = 0
        rightindex = n-1
        maxvolum = 0
        while leftindex < rightindex:
            maxvolum = max(maxvolum, (rightindex-leftindex) * min(height[leftindex], height[rightindex]))
            if height[leftindex] < height[rightindex]:
                leftindex += 1
            else:
                rightindex -= 1
        return maxvolum

# @lc code=end

